### PR TITLE
[Fix] Refactor LineGraphs to allow mixed units for network graphs

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

Remove the concept of maxUnit inside our LineGraph abstraction, and replace it with two optional props:
- formatData(), called by the internal formatData method. Replaces the functionality of maxUnit, so consumers now have to pass readableBytes or whatever they'd like to use to shrink the data to a manageable unit.
- formatTooltip(), called by the tooltip callback inside LineGraph. We need two methods bc units in a tooltip can be mixed (to account for smaller values).

Rearranged things to work with the new pattern.

Possible future enhancements:
- Quarantine Longview utilities inside the LV directory, and make Manager-specific copies of any needed utilities to reduce the square peg/round hole problem. Alternatively, generalize our data pipeline so we can put different units through it.
- Create `variants` for Longview graphs using the LongviewLineGraph abstraction. Maybe `base`, `mixedBytes`, and `mixedBits` to handle the three main cases.
- Nix the `formatData` prop and convert data to maxUnit before sending it to the graph. Use new methods, something like `scaleInBits` and `scaleInBytes` to format the tooltip. Something like:

```
scaleInBits = (baseUnit: NetworkUnit, value: number) => {
    if (value < 0.05) {
       convertBaseUnitToSmallerUnit(value)
.....
```

## Notes

Please compare LV and Linode graphs with Classic and Cloud production. Use different clients to try to work with a range of data.

### CODE REVIEW

File: packages/manager/src/features/Longview/LongviewDetail/DetailTabs/Apache/ApacheGraphs.tsx [hunk 1]
```diff
@@ -130,61 +152,61 @@ export const ApacheGraphs: React.FC<CombinedProps> = props => {
                     label: 'Waiting',
                     borderColor: 'transparent',
                     backgroundColor: theme.graphs.workers.waiting,
-                    data: _convertData(workersWaiting, start, end, formatData)
+                    data: _convertData(workersWaiting, start, end)
```
Comment (Jskobos) [inline packages/manager/src/features/Longview/LongviewDetail/DetailTabs/Apache/ApacheGraphs.tsx]: We do rounding at the graph level now (through the tooltip formatters) so these aren't necessary.

File: packages/manager/src/features/Longview/LongviewDetail/DetailTabs/Apache/ApacheGraphs.tsx [hunk 2]
```diff
@@ -206,21 +228,11 @@ export const ApacheGraphs: React.FC<CombinedProps> = props => {
...
-  if (value === null) {
-    return value;
-  }
-
-  // Round to 2 decimal places.
-  return Math.round(value * 100) / 100;
-};
-
-const formatNetwork = (value: number | null) => {
+export const kilobytesToBits = (value: number | null) => {
```
Comment (Jskobos) [inline packages/manager/src/features/Longview/LongviewDetail/DetailTabs/Apache/ApacheGraphs.tsx]: This is misnamed, should be toBytes.

File: packages/manager/src/features/Longview/LongviewDetail/DetailTabs/OverviewGraphs/DiskGraph.tsx [hunk 3]
```diff
@@ -52,7 +52,7 @@ export const DiskGraph: React.FC<CombinedProps> = props => {
       error={(!data.Disk && requestError) || error}
       loading={loading}
       subtitle={'ops/second'}
-      unit={'ops/second'}
+      unit={' ops/second'}
```
Comment (Jskobos) [inline packages/manager/src/features/Longview/LongviewDetail/DetailTabs/OverviewGraphs/DiskGraph.tsx]: This is awkward but it's the easiest way for the consumer to have control over whether or not a space is wanted between the value and the unit. Sometimes it looks best without (10MB), sometimes with (3 ops/second).

File: packages/manager/src/components/LineGraph/LineGraph.tsx [hunk 4]
```diff
@@ -371,35 +366,24 @@ export const metricsBySection = (data: Metrics): number[] => [
...
      * is called directly by chart.js
      */
     const dataset = data[t.datasetIndex];
     const label = dataset.label;
     const val = dataset.data[t.index][1] || 0;
-    const value = maxUnit
-      ? readableBytes(val).formatted
-      : Math.round(val * 100) / 100;
-    return `${label}: ${value} ${unit ? unit : ''}`;
+    const value = formatter ? formatter(val) : Math.round(val * 100) / 100;
```
Comment (johnwcallahan) [inline packages/manager/src/components/LineGraph/LineGraph.tsx]: Slight enhancement: move `Math.round(val * 100) / 100;` to its own function to keep this high-level, i.e:

```typescript
const value = formatter ? formatter(val) : roundToTwoDecimalPlaces(val);
```

File: packages/manager/src/features/Longview/LongviewDetail/DetailTabs/Apache/ApacheGraphs.tsx [hunk 5]
```diff
@@ -76,7 +80,19 @@ export const ApacheGraphs: React.FC<CombinedProps> = props => {
   const totalKBytes = data?.['Total kBytes'] ?? [];
   const totalAccesses = data?.['Total Accesses'] ?? [];
 
-  const unit = React.useMemo(() => getMaxUnit([totalKBytes]), [totalKBytes]);
+  /**
+   * NB: unlike some other places in Longview,
+   * totalKBytes appears to actually be returned
+   * in kilobytes. Our network helper utilities
```
Comment (johnwcallahan) [inline packages/manager/src/features/Longview/LongviewDetail/DetailTabs/Apache/ApacheGraphs.tsx]: Maybe add something to the effect of `"...actually be returned in kilobytes (as opposed to bytes)"`. This comment may be confusing without knowledge of the behavior of the other LV sections.

File: packages/manager/src/features/Longview/LongviewDetail/DetailTabs/Apache/ApacheGraphs.tsx [hunk 6]
```diff
@@ -109,14 +125,25 @@ export const ApacheGraphs: React.FC<CombinedProps> = props => {
             <Grid item xs={12} sm={6} className={classes.smallGraph}>
               <LongviewLineGraph
                 title="Throughput"
-                subtitle={`${unit}/s`}
-                maxUnit={unit as StorageSymbol}
+                subtitle={`${networkUnit}/s`}
+                unit={'/s'}
+                formatData={(value: number) =>
+                  Math.round(
```
Comment (johnwcallahan) [inline packages/manager/src/features/Longview/LongviewDetail/DetailTabs/Apache/ApacheGraphs.tsx]: Could probably use the "roundToTwoDecimalPlaces" fn here too. I know the snippet is well known and small, but our indentation makes it slightly jarring here on first glance.

File: packages/manager/src/features/linodes/LinodesDetail/LinodeSummary/NetworkGraph.tsx [hunk 7]
```diff
@@ -189,31 +190,34 @@ const Graph: React.FC<GraphProps> = props => {
...
+
+  /**
+   * formatNetworkTooltip is a helper method from Longview, where
+   * data is expected in bytes. The method does the rounding, unit conversions, etc.
+   * that we want, but it first multiplies by 8 to convert to bits.
+   * APIv4 returns this data in bits to begin with,
+   * so we have to preemptively divide by 8 to counter the conversion inside the helper.
+   *
+   */
+  const _formatTooltip = (value: number) => formatNetworkTooltip(value / 8);
```
Comment (johnwcallahan) [inline packages/manager/src/features/linodes/LinodesDetail/LinodeSummary/NetworkGraph.tsx]: Changing the argument name to `valueInBits` might help reinforce what you're saying in the comment.

**Pull request merged**